client/app/pages/queries/hooks/useAutoLimitFlags.js

```diff
@@ -0,0 +1,24 @@
+import { useCallback, useMemo, useState } from "react";
+import localOptions from "@/lib/localOptions";
+import { extend } from "lodash";
+
+function isAutoLimitAvailable(dataSource) {
+  return dataSource !== null && dataSource.apply_auto_limit;
```

Comment:
Just in case it can be `undefined`, let's go with either `!!dataSource && dataSource.apply_auto_limit`, or with the version I personally like more, with lodash `get(dataSource, "apply_auto_limit", false)`